### PR TITLE
 Document the Metrics provided Out of the Box

### PR DESCRIPTION
**Pull request by berngp**
This commit documents the metrics supported out of the box via the `graphql-dgs-spring-boot-micrometer` module.

----

This commit enables the [admonitions plugin](https://squidfunk.github.io/mkdocs-material/reference/admonitions/) as well as provide a `Makefile` that can be used, via Docker, to serve the site locally. To do so just type `make serve`.

### CODE REVIEW

File: docs/advanced/instrumentation.md [hunk 1]
```diff
@@ -121,35 +121,177 @@ If we wouldn't do this, the result for an async data fetcher would always be 0,
...
+        <dependency>
+            <groupId>com.netflix.graphql.dgs</groupId>
+            <artifactId>graphql-dgs-spring-boot-micrometer</artifactId>
+            <version>3.+</version>
+        <dependency>
+    </dependencies>
+    ```
+!!! warning
+    The version used above is just an example. Please verify the verison you want to use by visiting
+    the [Release page](https://github.com/Netflix/dgs-framework/release://github.com/Netflix/dgs-framework/releases).
```
Comment (srinivasankavitha) [inline docs/advanced/instrumentation.md:191]: Release page link doesn't work.

File: docs/advanced/instrumentation.md [hunk 2]
```diff
@@ -121,35 +121,177 @@ If we wouldn't do this, the result for an async data fetcher would always be 0,
...
+
+### Configuration
+
+#### management.metrics.dgs-graphql.enabled
+
+Enables the metrics provided out of the box.
+**Defaults** to `true`.
+
+#### management.metrics.dgs-graphql.tag-customizers.outcome.enabled
+
```
Comment (srinivasankavitha) [inline docs/advanced/instrumentation.md:279]: Maybe add a sentence or two clarifying that  for GraphQL, the success or failure is based on whether there were any graphql errors, even though the Http request by itself is succesful.

**Review by srinivasankavitha — CHANGES_REQUESTED** [review]

File: docs/advanced/instrumentation.md [hunk 3]
```diff
@@ -121,35 +121,183 @@ If we wouldn't do this, the result for an async data fetcher would always be 0,
...
+| `gql.loaderBatchSize` | The number of queries executed in the batch. |
+
+
+### Data Fetcher Timer: gql.resolver
+
+Captures the elapsed time of each data fetcher invocation.
+This is useful if you want to find data fetchers that might be responsible for poor query performance.
+
+!!! note
+    This metric is not available if the data is resolved via a Batch Loader.
```
Comment (paulbakker) [inline docs/advanced/instrumentation.md:243]: By default this is true. Previously you could override the behavior with an annotation, to explicitly enable/disable metrics on a fetcher. Is that still supported?

File: docs/advanced/instrumentation.md [hunk 4]
```diff
@@ -121,35 +121,183 @@ If we wouldn't do this, the result for an async data fetcher would always be 0,
...
+
+**Name:** `gql.resolver`
```
Comment (paulbakker) [inline docs/advanced/instrumentation.md:245]: Should there be a section about what is considered a fetcher? E.g. by default we record every custom defined `@DgsData`, but we don’t capture _trivial_ datafetchers (by design).

**Review by paulbakker — APPROVED** [review]
LGTM. Added to comments for extra sections, but everything that’s there looks good.

**Review by srinivasankavitha — APPROVED** [review]

File: docs/advanced/instrumentation.md (hunk 3 above)
Comment (berngp) [reply docs/advanced/instrumentation.md:243]: We don't supported in the OSS module, will open a ticket to add it.

File: docs/advanced/instrumentation.md (hunk 4 above)
Comment (berngp) [reply docs/advanced/instrumentation.md:252]: Added a small section to to this.
image

File: docs/advanced/instrumentation.md (hunk 3 above)
Comment (berngp) [reply docs/advanced/instrumentation.md]: Tracking via https://github.com/Netflix/dgs-framework/issues/251

**Pull request merged**